server/service/service_appconfig.go

```diff
@@ -319,3 +320,106 @@ func (svc *Service) SetupRequired(ctx context.Context) (bool, error) {
 	}
 	return false, nil
 }
+
+func (svc *Service) LoggingConfig(ctx context.Context) (*fleet.Logging, error) {
+	if err := svc.authz.Authorize(ctx, &fleet.AppConfig{}, fleet.ActionRead); err != nil {
+		return nil, err
+	}
+	conf := svc.config
+	logging := &fleet.Logging{
+		Debug: conf.Logging.Debug,
+		Json:  conf.Logging.JSON,
+	}
+
+	switch conf.Osquery.StatusLogPlugin {
+	case "", "filesystem":
+		logging.Status = fleet.LoggingPlugin{
+			Plugin: "filesystem",
+			Config: fleet.FilesystemConfig{FilesystemConfig: conf.Filesystem},
+		}
+	case "kinesis":
+		logging.Status = fleet.LoggingPlugin{
+			Plugin: "kinesis",
+			Config: fleet.KinesisConfig{
+				Region:       conf.Kinesis.Region,
+				StatusStream: conf.Kinesis.StatusStream,
+				ResultStream: conf.Kinesis.ResultStream,
+			},
+		}
+	case "firehose":
+		logging.Status = fleet.LoggingPlugin{
+			Plugin: "firehose",
+			Config: fleet.FirehoseConfig{
+				Region:       conf.Firehose.Region,
+				StatusStream: conf.Firehose.StatusStream,
+				ResultStream: conf.Firehose.ResultStream,
+			},
+		}
+	case "lambda":
+		logging.Status = fleet.LoggingPlugin{
+			Plugin: "lambda",
+			Config: fleet.LambdaConfig{
+				Region:         conf.Lambda.Region,
+				StatusFunction: conf.Lambda.StatusFunction,
+				ResultFunction: conf.Lambda.ResultFunction,
+			},
+		}
+	case "pubsub":
+		logging.Status = fleet.LoggingPlugin{
+			Plugin: "pubsub",
+			Config: fleet.PubSubConfig{PubSubConfig: conf.PubSub},
+		}
+	case "stdout":
+		logging.Status = fleet.LoggingPlugin{Plugin: "stdout"}
+	default:
+		return nil, errors.Errorf("unrecognized logging plugin: %+x", logging.Status)
```

Comment:
Not sure I understand the intent of using `%+x` and `logging.Status` here. Does this print an address to the `logging.Status`? I think it might be more helpful to print `conf.Osquery.StatusLogPlugin` as a string.

Reply:
Left is from debugging. Thanks for spotting it